Add qv photochemistry diagnostics

DWesl opened the pull request:
## Description

Adds arrays to accumulate the water vapor photochemistry terms and write them out.  

This PR should not change model output, but it will change water vapor diagnostics in the stratosphere and mesosphere to reflect changes from methane oxidation and photodissociation.
<!--
(Instructions: this, and all subsequent sections of text should be removed and filled in as appropriate.)
Provide a detailed description of what this PR does.  
What bug does it fix, or what feature does it add?  
Is a change of answers expected from this PR?  
-->

I should mention that one of the parameters gets called `dtend_qv_ozmix`, because the $-\frac{1}{\tau}(r-r_0)$ term it represents here is analagous to the $\left.\frac{\partial(P-L)}{\partial r}\right|_0 (r - r_0)$ ozone photochemistry diagnostic called `dtend_o3_ozmix`.  I don't know how to change the CCPP machinery to change the name of the diagnostic, nor what to change it to if I did.


### Issue(s) addressed

<!-- Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues should always be created before starting work on a PR branch!)
-->
- fixes ufs-community/ccpp-physics#255



## Testing
<!--
How were these changes tested?  
What compilers / HPCs was it tested with?  
Are the changes covered by regression tests? (If not, why? Do new tests need to be added?)  
Have the ufs-weather-model regression test been run? On what platform?  
- Will the code updates change regression test baseline? If yes, why? Please show the baseline directory below.
- Please commit the regression test log files in your ufs-weather-model branch
-->
- Tested on Hera with default global-workflow compilers, probably Intel
- There should not be changes outside diagnostic output, so existing regression tests should suffice
- I need to look up how to run the regression tests, and have not run them
- This should not require baseline updates, unless they track specific humidity diagnostics from the stratosphere or mesosphere

## Dependencies
<!--
If testing this branch requires non-default branches in other repositories, list them.
Those branches should have matching names (ideally)

Do PRs in upstream repositories need to be merged first?
If so add the "waiting for other repos" label and list the upstream PRs
- waiting on noaa-emc/nems/pull/<pr_number>
- waiting on noaa-emc/fv3atm/pull/<pr_number>
-->
- Depends on ufs-community/ccpp-physics#265 for the code to actually calculate the specific humidity tendencies

DWesl committed:
ENH: Add water vapor photochemistry diagnostics

DWesl committed:
ENH: Add water vapor photochemistry diagnostics

DWesl committed:
Ensure qv photochemistry diagnostics are only in runs with qv photochemistry.

DWesl committed:
Merge work from DWesl/fv3atm into add-qv-photochemistry-diagnostics

grantfirl commented:
@DWesl I don't see a ufs-weather-model pull request for this work. Are you going to add one?

DWesl committed:
Merge upstream develop into branch with required changes in CCPP-Physics

DWesl commented:
> @DWesl I don't see a ufs-weather-model pull request for this work. Are you going to add one?

Oh, right.  What branch would that be against?  `develop` is pulling in more FV3 changes than mine

grantfirl commented:
> > @DWesl I don't see a ufs-weather-model pull request for this work. Are you going to add one?
> 
> Oh, right. What branch would that be against? `develop` is pulling in more FV3 changes than mine

It needs to go into develop.

DWesl commented:
ufs-community/ufs-weather-model#2707 created to pull these changes into that repository.

DWesl committed:
Merge upstream updates into ccpp/physics

DWesl committed:
BUG: Avoid divide-by-zero in CCPP-Physics ozone diagnostic calculations

DWesl committed:
Merge CCPP-Physics updates from upstream

DWesl committed:
Merge branch 'develop' of NOAA-EMC/fv3atm into add-qv-photochemistry-diagnostics

DWesl committed:
Merge branch 'develop' of NOAA-EMC/fv3atm into add-qv-photochemistry-diagnostics

DWesl committed:
Merge branch 'develop' of NOAA-EMC/fv3atm into add-qv-photochemistry-diagnostics

DWesl committed:
BUG: Allow photochemistry schemes to have parameters at different times.

In particular, allow the ozone scheme to have two extra sets of parameters during the ozone hole without crashing the model.

DWesl committed:
Merge branch 'develop' of NOAA-EMC/fv3atm into photochemistry-index-fix

DWesl committed:
STY: Move the find_photochemistry_index calls inside the conditionals.

With a separate index for each photochemistry scheme, we can move the index-finding calls
in the conditional for the photochemistry scheme.

DWesl committed:
Merge branch 'develop' of NOAA-EMC/fv3atm into photochemistry-index-fix

DWesl committed:
Merge branch 'develop' of NOAA-EMC/fv3atm into add-qv-photochemistry-diagnostics

DWesl committed:
Merge photochemistry index fix into the photochemistry diagnostics addition.

That was ufs-community/ccpp-physics#286, with corresponding PR NOAA-EMC/fv3atm#971 here.

DWesl committed:
Merge branch 'photochemistry-index-fix' into add-qv-photochemistry-diagnostics

DWesl committed:
Merge branch 'develop' of NOAA-EMC/fv3atm into add-qv-photochemistry-diagnostics

DWesl committed:
Update CCPP-Physics with upstream changes

DWesl committed:
Merge branch 'develop' of NOAA-EMC/fv3atm into add-qv-photochemistry-diagnostics

DusanJovic-NOAA commented:
Merged via #990
